core/error.rs

```diff
@@ -97,23 +97,35 @@ pub fn to_v8_error<'a>(
   get_class: GetErrorClassFn,
   error: &Error,
 ) -> v8::Local<'a, v8::Value> {
-  let cb = JsRuntime::state(scope)
+  let tc_scope = &mut v8::TryCatch::new(scope);
+  let cb = JsRuntime::state(tc_scope)
     .borrow()
     .js_build_custom_error_cb
     .clone()
     .expect("Custom error builder must be set");
-  let cb = cb.open(scope);
-  let this = v8::undefined(scope).into();
-  let class = v8::String::new(scope, get_class(error)).unwrap();
-  let message = v8::String::new(scope, &format!("{:#}", error)).unwrap();
+  let cb = cb.open(tc_scope);
+  let this = v8::undefined(tc_scope).into();
+  let class = v8::String::new(tc_scope, get_class(error)).unwrap();
+  let message = v8::String::new(tc_scope, &format!("{:#}", error)).unwrap();
```

Comment:
Oh this is interesting. I didn't notice that we format the rust errors using the Debug trait. Debug traits might include sensitive information (crypto material, file paths, platform info) - we should be more careful with `type_error` and `generic_error`.

Reply:
I believe it is on purpose, otherwise the error includes only the first line which loses a lot of context in different scenarios. @dsherret we use that in Node compat IIRC?

Reply:
If it had a question mark it would be debug. The `{:#}` is "alternate" form of printing https://doc.rust-lang.org/std/fmt/#sign0 -- with anyhow it includes the context.